# Publish all joint states -----------------------------------
joint_state_controller:
  type: joint_state_controller/JointStateController
  publish_rate: 50

# Table push link joint controller --------------------------
push_link_controller:
  type: position_controllers/JointPositionController
  joint: push_link_joint
  pid: {p: 10.0, i: 10, d: 0.0}

# Controller for the rotating joints

left_back_leg_controller: 
  type: position_controllers/JointPositionController
  joint: left_back_leg_joint
  pid: {p: 10.0, i: 10, d: 0.0}

right_back_leg_controller: 
  type: position_controllers/JointPositionController
  joint: right_back_leg_joint
  pid: {p: 10.0, i: 10, d: 0.0}

left_front_leg_controller: 
  type: position_controllers/JointPositionController
  joint: left_front_leg_joint
  pid: {p: 10.0, i: 10, d: 0.0}

right_front_leg_controller: 
  type: position_controllers/JointPositionController
  joint: right_front_leg_joint
  pid: {p: 10.0, i: 10, d: 0.0}

# Velocity controllers ---------------------------------------
left_wheel_controller: 
  type: velocity_controllers/JointVelocityController
  joint: left_shaft_joint
  pid: { p: 100.0, i: 1, d: 10}

right_wheel_controller: 
  type: velocity_controllers/JointVelocityController
  joint: right_shaft_joint
  pid: { p: 100.0, i: 1, d: 10}

# Use the diff drive controller ------------------------------
mobile_base_controller:
  type        : "diff_drive_controller/DiffDriveController"
  left_wheel  : 'left_shaft_joint'
  right_wheel : 'right_shaft_joint'
  publish_rate: 50.0               # default: 50
  pose_covariance_diagonal : [0.001, .001, 1000000., 1000000., 1000000., 1000.]
  twist_covariance_diagonal: [0.001, .001, 1000000., 1000000., 1000000., 1000.]
 
  # Wheel separation and diameter. These are both optional.
  # diff_drive_controller will attempt to read either one or both from the
  # URDF if not specified as a parameter
  wheel_separation : 0.65
  wheel_radius : 0.075
 
  # Wheel separation and radius multipliers
  wheel_separation_multiplier: 1.0 # default: 1.0
  wheel_radius_multiplier    : 1.0 # default: 1.0
 
  # Velocity commands timeout [s], default 0.5
  cmd_vel_timeout: 1.0
 
  # Base frame_id
  base_frame_id: base_link #default: base_link
 
  # Velocity and acceleration limits
  # Whenever a min_* is unspecified, default to -max_*
  linear:
    x:
      has_velocity_limits    : true
      max_velocity           : 1.0  # m/s
      min_velocity           : -1.0 # m/s
      has_acceleration_limits: true
      max_acceleration       : 0.8  # m/s^2
      min_acceleration       : -0.8 # m/s^2
      has_jerk_limits        : true
      max_jerk               : 5.0  # m/s^3
  angular:
    z:
      has_velocity_limits    : true
      max_velocity           : 4.0  # rad/s
      has_acceleration_limits: true
      max_acceleration       : 2.0  # rad/s^2
      has_jerk_limits        : true
      max_jerk               : 2.5  # rad/s^3